### PR TITLE
move waiting cluster ready to a step

### PR DESCRIPTION
**Pull request by yasun1**
On Prow CI, the shared files are written to SHARED_DIR at the end, if the step is interrupted in the middle, e.g. waiting for the cluster to be ready, there are no cluster-id/cluster-name files so the deprovision step will not be executed. Hence, move the waiting process to an individual step. [OCPQE-20632](https://issues.redhat.com//browse/OCPQE-20632)

Besides, the `-m auto` is only available for the `sts` cluster, and if `BYO_OIDC` is set to true, we should choose the auto model too. [OCPQE-20633](https://issues.redhat.com//browse/OCPQE-20633)

### CODE REVIEW

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-aws-rosa-hcp-stage-critical-f14 periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-aws-rosa-non-sts-byo-vpc-kms-proxy-f14 
periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-aws-rosa-sts-byo-kms-oidc-f14 periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-aws-rosa-sts-shared-vpc-f14 periodic-ci-openshift-openshift-tests-private-master-rosacli-aws-rosa-shared-vpc-auto-e2e-f7

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-aws-rosa-sts-byo-kms-oidc-f14 periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-aws-rosa-sts-shared-vpc-f14 periodic-ci-openshift-openshift-tests-private-master-rosacli-aws-rosa-shared-vpc-auto-e2e-f7 periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-aws-rosa-hcp-byo-kms-oidc-auditlog-stage-critical-f14

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-aws-rosa-hcp-stage-critical-f14 periodic-ci-openshift-openshift-tests-private-master-rosacli-aws-rosa-shared-vpc-auto-e2e-f7 private-release-4.15-amd64-stable-aws-rosa-hcp-byo-kms-oidc-auditlog-stage-critical-f14

**Comment by yasun1** [comment]
/pj-rehearse  periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-aws-rosa-hcp-byo-kms-oidc-auditlog-stage-critical-f14

**Comment by yasun1** [comment]
@xueli181114 @yunjiang29 Please help review. Thanks.

File: ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh [hunk 1]
```diff
@@ -514,180 +520,21 @@ ${SHARED_VPC_SWITCH} \
 ${SECURITY_GROUP_ID_SWITCH} \
 ${NO_CNI_SWITCH} \
 ${DRY_RUN_SWITCH}
-" | sed -E 's/\s{2,}/ /g' > "${SHARED_DIR}/create_cluster.sh"
+" 
+echo "$cmd"| sed -E 's/\s{2,}/ /g' > "${SHARED_DIR}/create_cluster.sh"
 cat "${SHARED_DIR}/create_cluster.sh" | sed "s/$AWS_ACCOUNT_ID/$AWS_ACCOUNT_ID_MASK/g" > "${ARTIFACT_DIR}/create_cluster.sh"
```
Comment (yunjiang29) [inline ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh:525]: it's a little confused between `${SHARED_DIR}/create_cluster.sh` and `${ARTIFACT_DIR}/create_cluster.sh`, suggest to 
```suggestion
cat "${SHARED_DIR}/create_cluster.sh" | sed "s/$AWS_ACCOUNT_ID/$AWS_ACCOUNT_ID_MASK/g" > "${ARTIFACT_DIR}/create_cluster_command_for_debugging.txt"
```
or something  like that?

Comment (yasun1) [reply ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh:525]: Seems like we do not need `${ARTIFACT_DIR}/create_cluster.sh` for we have print it in the terminal.

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-aws-rosa-shared-vpc-auto-e2e-f7

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-aws-rosa-shared-vpc-auto-e2e-f7

**Comment by yunjiang29** [comment]
looks good to me

**Pull request merged**